osu.Game/Overlays/DirectOverlay.cs

```diff
@@ -179,6 +179,41 @@ private void updateResultCounts()
                                     "Tag".ToQuantity(ResultAmounts.Tags);
         }
 
+        private IEnumerable<DirectPanel> createPanels(PanelDisplayStyle displayStyle, IEnumerable<BeatmapSetInfo> sets)
+        {
+            return sets.Select<BeatmapSetInfo, DirectPanel>(b =>
+            {
+                switch (displayStyle)
+                {
+                    case PanelDisplayStyle.Grid:
+                        return new DirectGridPanel(b)
+                        {
+                            Anchor = Anchor.TopCentre,
+                            Origin = Anchor.TopCentre,
+                        };
+
+                    default:
+                        return new DirectListPanel(b);
+                }
+            });
+        }
+
+        private void addPanels(PanelDisplayStyle displayStyle, IEnumerable<BeatmapSetInfo> sets)
+        {
+            if (BeatmapSets == null)
+                return;
+
+            var newPanels = createPanels(displayStyle, sets);
+
+            LoadComponentsAsync(newPanels, p =>
+            {
+                if (panels == null)
+                    return;
```

Comment:
Guaranteed non-null if I'm not missing anything.

Reply:
I think that the original idea with this check had to do with the fact that `recreatePanels` temporarily sets `panels` to null before recreating them in a `LoadComponentAsync` call. The only time I could see an issue here is if `recreatePanels` was called as a result of the display style being updated while adding additional panels. Do you think the guard should be dropped?

Reply:
So I did miss something. Would be nice to get rid of it, but I suppose it can stay.